polymath/extensions/vector_ops.py

```
@@ -4,691 +4,710 @@
+            factor = arg.wod / obj
+            for key, arg_deriv in arg._derivs.items():
+                obj.insert_deriv(key, Qube.dot(factor, arg_deriv, a1, a1, classes=classes,
+                                               recursive=False))
 
-        obj.insert_derivs(new_derivs)
+        return obj
 
-    return obj
+    @staticmethod
+    @staticmethod
+    def norm_sq(arg, axis=-1, *, classes=(), recursive=True):
```

Ruff (0.14.7)

Missing return type annotation for staticmethod `norm_sq`

(ANN205)